src/GraphQL/Types/Collections/SchemaTypes.cs

```diff
@@ -127,8 +142,12 @@ protected void Initialize(ISchema schema, IServiceProvider serviceProvider)
             _initialized = true;
 
             var types = GetSchemaTypes(schema, serviceProvider);
-            var typeMappingsEnumerable = schema.TypeMappings ?? throw new ArgumentNullException(nameof(schema) + "." + nameof(ISchema.TypeMappings));
-            var typeMappings = typeMappingsEnumerable is List<(Type, Type)> typeMappingsList ? typeMappingsList : typeMappingsEnumerable.ToList();
+            if (schema.TypeMappings != null)
+            {
+                // this code could be moved into Schema
+                var additionalMappings = schema.TypeMappings.Select(x => new ManualGraphTypeMapping(x.clrType, x.graphType));
+                graphTypeMappings = graphTypeMappings != null ? graphTypeMappings.Concat(additionalMappings).ToList() : additionalMappings.ToList();
```

Comment:
```suggestion
                var manualMappings = schema.TypeMappings.Select(x => new ManualGraphTypeMapping(x.clrType, x.graphType));
                graphTypeMappings = graphTypeMappings != null ? graphTypeMappings.Concat(manualMappings).ToList() : additionalMappings.ToList();
```

Reply:
Also I think manualMappings should go first for `Concat`.

Reply:
I reversed it so the manual mappings would always take precedence over DI-configured mappings.  (Last-wins with this new design versus first-wins before.)

Reply:
Where?

Reply:
> I reversed it

Where is the priority of manual mappings over other ones?

Reply:
See `GetGraphTypeFromClrType`:

```cs
            Type? mappedType = null;

            // check custom mappings first
            if (typeMappings != null)
            {
                foreach (var mapping in typeMappings)
                {
                    mappedType = mapping.GetGraphTypeFromClrType(clrType, isInputType, mappedType);
                }
            }

            if (mappedType != null)
                return mappedType;
```

In this code, for any `ManualGraphTypeMappingProvider`, the LAST registration in `typeMappings` above is going to take effect.  So `SchemaTypes.Initialize` puts manual registrations in the schema (`schema.TypeMappings`) LAST in the enumerable by using concat:

```cs
            if (schema.TypeMappings != null)
            {
                // this code could be moved into Schema
                var additionalMappings = schema.TypeMappings.Select(x => new ManualGraphTypeMappingProvider(x.clrType, x.graphType));
                graphTypeMappings = graphTypeMappings != null ? graphTypeMappings.Concat(additionalMappings).ToList() : additionalMappings.ToList();
            }
```

Does this not make sense?

Reply:
OK. I see, it does. I'm fine with excessive call to `MakeGenericType`.

Reply:
Just FYI, for any ManualGraphTypeMappingProvider registrations, there are no calls to MakeGenericType at all.  And for `AutoRegisteringGraphTypeMappingProvider`, if a prior registration has set the type, it also will not call MakeGenericType.  And if `AutoRegisteringGraphTypeMappingProvider` does return a type, it will be returned by `GetGraphTypeFromClrType` prior to any other code.  So really I cannot see how (with the included implementations) there will be multiple calls to `MakeGenericType`.

Reply:
not multiple but excessive one

```
 var additionalMappings = schema.TypeMappings.Select(x => new ManualGraphTypeMappingProvider(x.clrType, x.graphType));
graphTypeMappings = graphTypeMappings != null ? graphTypeMappings.Concat(additionalMappings).ToList() : additionalMappings.ToList();
```

`graphTypeMappings` here may contain `AutoRegisteringGraphTypeMappingProvider` so it goes before manual mapping. First `AutoRegisteringGraphTypeMappingProvider` always creates some type via `MakeGenericType`, then this type flows into `ManualGraphTypeMappingProvider` that _may_ discard it, not always of course. I'm not against such behaviour. OK to keep as is.

Reply:
You’re right.